sherpa/astro/xspec/tests/test_xspec_unit.py

```diff
@@ -826,38 +827,68 @@ def test_get_xsstate_keys():
 
 
 @requires_xspec
-def test_set_xsstate_missing_key():
-    """Check set_xsstate does nothing if required key is missing.
+@pytest.mark.parametrize("key",
+                         ["abund", "chatter", "cosmo", "xsect",
+                          "modelstrings"])  # paths is not a required key
+def test_set_xsstate_missing_key(key):
+    """Check set_xsstate handles a missing key.
 
     """
 
     from sherpa.astro import xspec
 
     ostate = xspec.get_xsstate()
-
+    assert key in ostate
     for val in ostate.values():
         assert val is not None
 
-    # paths is not a required key
+    def not_elem(value, vals):
+        """Pick the first item in vals that is not value"""
+        return [v for v in vals if v != value][0]
+
+    # We need valid arguments as the values will get set with the
+    # set_xsstate(fake) call below.
     #
-    req_keys = ["abund", "chatter", "cosmo", "xsect",
-                "modelstrings"]
-
-    fake = {'abund': ostate['abund'] + '_copy',
-            'xsect': ostate['xsect'] + '_copy',
-            'chatter': -10,
-            'cosmo': (0.0, 0.0),  # two elements will cause a failure
-            'modelstrings': {'foo': 2, 'bar': None},
+    fake = {'abund': not_elem(ostate['abund'], ["angr", "aspl", "grsa"]),
+            'xsect': not_elem(ostate['xsect'], ["bcmc", "vern", "obcm"]),
+            'chatter': 10,
+            'cosmo': (50, 0.1, 0.4),
+            'modelstrings': {'foo': '2'},
             'paths': {'manager': '/dev/null'}}
 
-    for key in req_keys:
+    del fake[key]
 
-        copy = fake.copy()
-        del copy[key]
-        xspec.set_xsstate(copy)
+    try:
+        xspec.set_xsstate(fake)
 
         nstate = xspec.get_xsstate()
-        assert nstate == ostate
+        ncopy = nstate.copy()
+        del ncopy[key]
+
+        # The key should be unchanged; the others set to those in
+        # fake. Checking the latter is a bit annoying.
+        #
+        assert nstate[key] == ostate[key]
+        for key, value in fake.items():
```

Comment:
argh - this over-writes the input `key` argument. Not a huge issue as such here but it needs cleaning up.

Reply:
This was found when working no some follow-up code in #2216 but worth cleaning up here.

Reply:
Addressed by commit f002119